File: examples-per-language/php/creational/Prototype.php
<?php

use designPatternsForHumans\creational\Prototype\Sheep;

require_once __DIR__ . '/autoload.php';

$original = new Sheep('Jolly');
echo $original->getName() . PHP_EOL;
echo $original->getCategory() . PHP_EOL;

$cloned = clone $original;
$cloned->setName('Dolly');
echo $cloned->getName() . PHP_EOL;
echo $cloned->getCategory() . PHP_EOL;

echo 'Remember, you can use the magic method __clone to modify the behaviour of cloning the object!';

File: Python/Creational/Prototype.py
import copy

class Sheep:
    _name = None
    _category = None

    def __init__(self, name, category = 'Mountain Sheep'):
        self.name = name
        self.category = category

    def setName(self, name):
        self.name = name

    def getName(self):
        return self.name

    def setCategory(self, category):
        self.category = category

    def getCategory(self):
        return self.category


if __name__ == '__main__':
    original = Sheep('Jolly')
    print original.getName()
    print original.getCategory()

    cloned = copy.copy(original)
    cloned.setName('Dolly')
    print cloned.getName()
    print cloned.getCategory()
    print original.getName()